spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessorTests.java

```diff
@@ -122,6 +123,54 @@ void testCloudSqlDataSourceWithIgnoredProvidedUrl() {
             });
   }
 
+  void testCloudSqlSpringDatasourceWithAllOptions() {
```

Comment:
Perhaps we could have one test that verifies the default values when these properties aren't customized?

Reply:
I think this behavior adequately tested in  `testCloudSqlDataSource()` test case.  

If a property is not set, the corresponding value in GcpCloudSqlProperties is `null`, and the JDBC URL property is not included in the final JDBC URL. 

Defaults are managed by the Cloud SQL JDBC Socket Factory library. We should not also manage defaults in the spring autoconfigure code.